packages/babel-parser/src/tokenizer/context.js

```diff
@@ -1,8 +1,7 @@
 // @flow
 
-// The token context is used to track whether `}` matches
-// a template quasi `${` or other tokens containing `{`:
-// namely tt.braceL `{` and tt.braceHashL `#{`
+// The token context is used to track whether the apostrophe "`"
+// starts or ends a string template
```

Comment:
I wonder if we could completely remove context and just use a `this.state.inTemplateLiteral` boolean flag, that we manage with `oldInTemplateLiteral = this.state.inTemplateLiteral; this.state.inTemplateLiteral = true` similarly to how we manage the other boolean variables.

Reply:
We are close to remove the whole token context. However the token context also records `preserveSpace`, to instruct whether tokenizer should skip space in `nextToken`. This feature is also used in JSX expressions so it can't be replaced by `state.inTemplateLiteral`.